### ISSUE TITLE
OpenFOAM-Extend/4.1-intel-2018a fails during sanity checking

### ISSUE THREAD

**Issue by moravveji** [easybuilders/easybuild-easyconfigs#9394]
Hi.

I adapted an existing easyconfig file for OpenFOAM-Extend version 4.0 to install version 4.1, using intel/2018a toolchain. However, the build fails during sanity checking with the following error:

`== FAILED: Installation ended unsuccessfully (build directory: /MyApps/2018a/software/OpenFOAM-Extend/4.1-intel-2018a): build failed (first 300 chars): Sanity check failed: no file found at 'foam-extend-4.1/applications/bin/linux64IccDPOpt/buoyantSimpleFoam' in /MyApps/2018a/software/OpenFOAM-Extend/4.1-intel-2018a
no file found at 'foam-extend-4.1/applications/bin/linux64IccDPOpt/buoyantBoussinesqSimpleFoam' in /MyApps/20
== Results of the build can be found in the log file(s) /tmp/eb-WFr_Sw/easybuild-OpenFOAM-Extend-4.1-20191127.161311.JAFir.log
ERROR: Build of /myData/easybuild/work/OpenFOAM-Extend/OpenFOAM-Extend-4.1-intel-2018a.eb failed (err: "build failed (first 300 chars): Sanity check failed: no file found at 'foam-extend-4.1/applications/bin/linux64IccDPOpt/buoyantSimpleFoam' in /MyApps/2018a/software/OpenFOAM-Extend/4.1-intel-2018a\nno file found at
'foam-extend-4.1/applications/bin/linux64IccDPOpt/buoyantBoussinesqSimpleFoam' in /MyApps/20")
`

What surprises me is the truncated installation path, which shows itself in the error message. I mean, the error message points to `/MyApps/20` instead of the usual path `/MyApps/2018a/software/OpenFOAM-Extend/4.1-intel-2018a`

Do you have an idea what could have gone wrong?
I have attached the adapted .eb file and the logs (after trimming the full paths to few locations).

Thanks in advance for your support.
[logs.tar.gz](https://github.com/easybuilders/easybuild-easyconfigs/files/3900481/logs.tar.gz)

**Comment by boegel** [comment]
The truncated path is a bit misleading, that only pops up because only the `first 300 chars` of the error message are shown.

When the sanity check fails to find a bunch of binaries like this, it usually indicates that the build went horrible wrong at some point. In that case, looking for `Error 1` or `error: ` in the build log usually helps to target the problem.

In this case the error that pops up is:

```
/myApps/software/OpenFOAM-Extend/4.1-intel-2018a/foam-extend-4.1/src/foam/lnInclude/meshObjectBase.H(70):
error: no suitable user-defined conversion from "Foam::HashTable<Foam::meshObjectBase *, Foam::word, Foam::string::hash>" to "Foam::HashTable<const Foam::meshObjectBase *, Foam::word, Foam::string::hash>" exists
                  mesh.objectRegistry::template lookupClass<meshObjectBase>();
                  ^
```

which then leads to:

```
ld: cannot find -ltopoChangerFvMesh
 make: *** [/myApps/software/OpenFOAM-Extend/4.1-intel-2018a/foam-extend-4.1/lib/linux64IccDPInt32Opt/libloadBalanceFvMesh.so] Error 1
```

In short, a trivial version bump isn't going to be possible here.
Unfortunately EasyBuild isn't smart enough yet to auto-fix broken code...

You either need a newer compiler, or perhaps a patch to fix the troublesome C++ code (or perhaps both)...

**Comment by moravveji** [comment]
Thanks Kenneth for your verdicts. Our latest stable toolchain is 2018a, and I have no patch file to apply to v. 4.0.

I am closing this ticket, and I will start looking in building it from source.

Thanks for your support

**Comment by boegel** [comment]
@moravveji I'm hitting the same problem now with `intel/2019b`, while looking into an installation request...

Did you ever fix this issue?

**Comment by boegel** [comment]
I've also asked the developers about this, see https://sourceforge.net/p/foam-extend/tickets/43/

**Comment by moravveji** [comment]
Hi Kenneth; In fact, I never made it, and eventually gave it, because it seemed to be an issue deep rooted in the source code, which I refrain from touching.
